src/wazuh_db/wdb_agent.c

```diff
@@ -564,41 +496,21 @@ int wdb_set_agent_status(int id_agent, int status) {
         return -1;
     }
 
-    if (wdb_open_global() < 0)
-        return -1;
-
-    if (wdb_prepare(wdb_global, SQL_UPDATE_AGENT_STATUS, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        return -1;
-    }
-
-    sqlite3_bind_text(stmt, 1, str_status, -1, NULL);
-    sqlite3_bind_int(stmt, 2, id_agent);
-
-    result = wdb_step(stmt) == SQLITE_DONE ? (int)sqlite3_changes(wdb_global) : -1;
-    sqlite3_finalize(stmt);
+    snprintf(wdbquery, OS_BUFFER_SIZE, global_db_queries[SQL_UPDATE_AGENT_STATUS], str_status, id_agent);
+    result = wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
 
-    return result;
+    return result == 0 ? 1 : -1;
```

Comment:
Log information about the result and make use of `OS_SUCCESS` and `OS_INVALID`

Reply:
Done!